pkg/controller/controller.go

```diff
@@ -392,15 +413,56 @@ func (ctrl *resizeController) markPVCAsFSResizeRequired(pvc *v1.PersistentVolume
 	newPVC := pvc.DeepCopy()
 	newPVC.Status.Conditions = util.MergeResizeConditionsOfPVC(newPVC.Status.Conditions,
 		[]v1.PersistentVolumeClaimCondition{pvcCondition})
-	_, err := util.PatchPVCStatus(pvc, newPVC, ctrl.kubeClient)
-	if err != nil {
+
+	if _, err := ctrl.patchPVCStatus(pvc, newPVC); err != nil {
 		klog.Errorf("Mark PVC %q as file system resize required failed: %v", util.PVCKey(pvc), err)
 		return err
 	}
-
 	klog.V(4).Infof("Mark PVC %q as file system resize required", util.PVCKey(pvc))
 	ctrl.eventRecorder.Eventf(pvc, v1.EventTypeNormal,
 		util.FileSystemResizeRequired, "Require file system resize of volume on node")
 
-	return err
+	return nil
+}
+
+func (ctrl *resizeController) patchPVCStatus(pvc, newPVC *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
+	key, err := getPVCKey(pvc)
+	if err != nil {
+		return nil, fmt.Errorf("generate unique key failed: %v", err)
+	}
+	// If pvc's ResourceVersion is equal to last processed ResourceVersion, it means this is a resync event,
+	// so we need to check PVC's version when patch its status to avoid double process on same PVC.
+	// More details see: https://github.com/kubernetes-csi/external-resizer/issues/4.
+	needVersionCheck := ctrl.lastProcessedPVCVersions.Get(key) == pvc.ResourceVersion
```

Comment:
I meant, lets not do all this stuff. Just add resource version to PVC and PV patch unconditionally. Let the patch operation fail and retry if we had stale object.

Reply:
Oh, got it. Updated.